### ISSUE TITLE
Better HTTP error handling

### ISSUE THREAD

**Issue by eed3si9n** [foundweekends/conscript#98]
e1d14c414ec7c53cc5df181b222f96f9f70a20ee changed all non-404 errors to display:

```scala
     """An unexpected error occurred: Please check your internet connection.
        |You may need to generate a github access token.
        |see https://help.github.com/articles/creating-an-access-token-for-command-line-use/
        |Exception message: %s""".stripMargin.format(e.getMessage)
```

Ref #69, #70

This is equally misleading as the original "And if you're using a pre-release version of Java, please try with a final release version."

### original comment from #69 by @michalmela

Maybe the HTTP error status should be checked against a list of known possible problems and appropriate message then shown?
Then if there is no known problem the error message could advise to re-visit documentation or sth.

What I mean is, the current message advises to set up the gh access token (it doesn't say where to put it which led me to this page exactly, but maybe I've not been reading the docs carefully enough), but this may only be helpful if the HTTP status was 403, as shown in @jisaacstone's quote:

> Exception message: Unexpected response status: 403

In my case the error was 301, caused by the fact that giter8 was apparently moved from n8han/giter8 to foundweekends/giter8 , so the new error message is not really helpful.

**Comment by jerzygangi** [comment]
A note to future Googlers:

If you come here after having run `n8han/giter8`, you need to instead run `foundweekends/giter8` since the repository was moved.

**Comment by eed3si9n** [comment]
Fixed by #102
